### ISSUE TITLE
examples not using reactor.run() should point to twistd

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10482]
|<img alt="dpollet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| dpollet reported|
|-|-|
|Trac ID|trac#754|
|Type|enhancement|
|Created|2004-10-20 21:32:35Z|
|Branch|https://github.com/twisted/twisted/tree/reactor-run-samples-754|

<details><summary>Searchable metadata</summary>

```
trac-id__754 754
type__enhancement enhancement
reporter__dpollet dpollet
priority__low low
milestone__ 
branch__branches_reactor_run_samples_754 branches/reactor-run-samples-754
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1098307955000000 1098307955000000
changetime__1218999098000000 1218999098000000
version__ 
owner__ 
cc__spiv cc__dpollet cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="dpollet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|dpollet commented|
|-|-|

```
#!html
<pre>
Most examples not using reactor.run() (eg. doc/examples/emailserver.py) just
create a TCPServer and add it to an application. Maybe they lack a comment
pointing to twistd if they won't run by just loading them with python ?

In fact what I was initially searching for was a way to use reactor.callLater
while having a TCPserver running... had to ditch the code using reactor and
setup a TimerService instead, using mktap and twistd. From my beginner point of
view, the two approaches look incompatible, which feels strange...

-- cDlm from #twisted
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
FWIW, I've already fixed doc/mail/examples/emailserver.py (it's location in SVN)
to have a short comment explaining how to run it.

It's probably worth looking at each file in doc/, though, and checking that all
valid TAC files (i.e. ones that can be used with twistd -y) are clearly marked
as such for newbies.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [24569]) Branching to 'reactor-run-samples-754'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix set owner to @thijstriemstra|
|-|-|

The only problem is the use of the word "module". Since they're .tac files, they're not importable, therefore they're not modules. Can you change it to ".tac file" or "file"?

Once you do that, feel free to merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [radix](#note_7):
> The only problem is the use of the word "module". Since they're .tac files, they're not importable, therefore they're not modules. Can you change it to ".tac file" or "file"?
>
> Once you do that, feel free to merge.

Alright, I changed it to '.tac file' in r24615

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24616]) Merge reactor-run-samples-754: adding a note for newbies to all .tac samples, explaining how to run the file with twistd.

Author: thijs
Reviewer: radix
Fixes: [#10482](https://github.com/twisted/twisted/issues/10482)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
